### PR TITLE
feat(loading-screen-for-root-app): loading screen for the root app

### PR DESCRIPTION
**Pull request by jerensl**
**Notes for Reviewers**

This loading is rendered on servers that don't have direct access to this current MUI version configuration. Also because the server doesn't have access to the react state for example useState, useEffect, etc, so I need to remove some of the animation

MUI in SSR/SSG supported in NextJS 13 or above
https://mui.com/material-ui/integrations/nextjs/#pages-router

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
Thanks!

**Comment by aabidsofi19** [comment]
@jerensl  while this is good , it would be better if this loading is also uplifted a bit , i.e in the root html while we are fetching the app.js / indexjs  we can also show the loading screen

File: ui/pages/_app.js [hunk 1]
```diff
@@ -681,105 +684,109 @@ class MesheryApp extends App {
...
-                        <MesheryProgressBar />
-                        {!this.state.isFullScreenMode && (
-                          <Header
-                            onDrawerToggle={this.handleDrawerToggle}
-                            onDrawerCollapse={isDrawerCollapsed}
-                            contexts={this.state.k8sContexts}
-                            activeContexts={this.state.activeK8sContexts}
-                            setActiveContexts={this.setActiveContexts}
-                            searchContexts={this.searchContexts}
+            {this.state.isLoading ? (
```
Comment (aabidsofi19) [inline ui/pages/_app.js:687]: please dont use a terninary operator here . it makes the whole block messy and hard to read . create a Loading component that take children and isLoading as props and renderes accordingly

**Review by aabidsofi19 — CHANGES_REQUESTED** [review]

File: ui/pages/_app.js [hunk 2]
```diff
@@ -681,105 +684,109 @@ class MesheryApp extends App {
...
+                    </div>
                   </div>
-                </div>
-                <PlaygroundMeshDeploy
-                  closeForm={() => this.setState({ isOpen: false })}
-                  isOpen={this.state.isOpen}
-                />
-              </ErrorBoundary>
-            </NoSsr>
+                  <PlaygroundMeshDeploy
```
Comment (leecalcote) [inline ui/pages/_app.js:783]: We shouldn’t have a component named playground.

**Comment by jerensl** [comment]
> @jerensl while this is good , it would be better if this loading is also uplifted a bit , i.e in the root html while we are fetching the app.js / indexjs we can also show the loading screen
> 
> said that it would need to be a plain html/css loading as we wont be able to access react there

I think it is fine to put in `_app.js` because the rendering happened in build time also we need to utilize loading state from `componentDidMount`, so far I have tested it in multiple pages, and I'm never seeing a blank white page

**Pull request merged**